src/http.cpp

```diff
@@ -269,22 +269,22 @@ int Exiv2::http(Exiv2::Dictionary& request, Exiv2::Dictionary& response, std::st
 
     ////////////////////////////////////
     // open the socket
-    int sockfd = (int)socket(AF_INET, SOCK_STREAM, IPPROTO_TCP);
+    int sockfd = socket(AF_INET, SOCK_STREAM, IPPROTO_TCP);
     if (sockfd < 0)
         return error(errors, "unable to create socket\n", nullptr, nullptr, 0);
 
     // fill in the address
     struct sockaddr_in serv_addr;
     int serv_len = sizeof(serv_addr);
-    memset((char*)&serv_addr, 0, serv_len);
+    memset(reinterpret_cast<char*>(&serv_addr), 0, serv_len);
 
     serv_addr.sin_addr.s_addr = inet_addr(servername_p);
     serv_addr.sin_family = AF_INET;
     serv_addr.sin_port = htons(atoi(port_p));
 
     // convert unknown servername into IP address
     // http://publib.boulder.ibm.com/infocenter/iseries/v5r3/index.jsp?topic=/rzab6/rzab6uafinet.htm
-    if (serv_addr.sin_addr.s_addr == (unsigned long)INADDR_NONE) {
+    if (serv_addr.sin_addr.s_addr == static_cast<unsigned long> INADDR_NONE) {
```

Comment:
[blocker] There is a typo here. `static_cast<>` is missing the parenthesis around `INADDR_NONE`. Interestingly this was only caught on CI by the OSX builds:

```
/Users/travis/build/Exiv2/exiv2/src/http.cpp:287:64: error: expected '(' after 'static_cast'
    if (serv_addr.sin_addr.s_addr == static_cast<unsigned long> INADDR_NONE) {
```

Reply:
Compiled fine locally. I’ll fix as soon as I can.

Reply:
Wow.  This is quite a fine.  On Monday I was investigating an ASAN is with gethostbyname() on the next line of code.  I haven't decide yet what to do about this yet, however it's getting thought and attention.

I'll deal with ASAN on 0.27-maintenance.  We only started to exercise this code in the test suite with 0.27.3

```
    serv_addr.sin_addr.s_addr   = inet_addr(servername_p);
    serv_addr.sin_family        = AF_INET    ;
    serv_addr.sin_port          = htons(atoi(port_p));

    // convert unknown servername into IP address
    // http://publib.boulder.ibm.com/infocenter/iseries/v5r3/index.jsp?topic=/rzab6/rzab6uafinet.htm
    if (serv_addr.sin_addr.s_addr == (unsigned long)INADDR_NONE)
    {
        struct hostent* host = gethostbyname(servername_p);   // <-- ASAN when executed from python, OK from bash 
        if ( !host )  return error(errors, "no such host", servername_p);
        memcpy(&serv_addr.sin_addr,host->h_addr,sizeof(serv_addr.sin_addr));
    }
```

Reply:
I think I figured out why it was compiling...

There are two )) that are needed at the end. Basically, static_cast<unsigned long> INADDR_NONE were evaluated separately.

Reply:
I've logged my RemoteIo/ASAN discoveries: #1307